### ISSUE TITLE
bisection num_steps = 1 solves the "guess" problem, not the final problem

### ISSUE THREAD

**Issue by msparapa** [Rapid-Design-of-Systems-Laboratory/beluga#140]
If `num_steps = 1`, then the bisection strategy should try and solve the final problem as opposed to the guess problem generated. That's not how it's currently set up though.

**Comment by msparapa** [comment]
If the keyword arg `steps` into `beluga.solve()` is `None`, then beluga will correctly assume it needs to solve the final problem as defined. This issue is not fixed for the specific case where the `bisection` strategy only has 1 step, however.

**Comment by SeanMatthewNolan** [comment]
I've admittedly known about this for years. I used it to look at the "guess," but as it isn't what would be expected, it probably should go.

**Comment by msparapa** [comment]
Agreed. I made a change where if `steps=None`, then beluga will try to solve the final problem as defined and not the guess. I think with bisection `num_steps=1` should pretty much follow the exact same behavior. This might ultimately mean that the converging to initial guess step should be its own isolated step, separate from the continuation strategies altogether.

**Comment by SeanMatthewNolan** [comment]
Thinking about it, the first step never actually makes progress. For example, if number of cases are set to 21, 20 are actually taken with the first "step" re-solving the previous problem with new scaling. If this step is actually needed, maybe we shift the number of cases by 1. In that way, you could theoretically enter 0 and get the past behavior, and everything makes more sense.
